src/ldp.rs

```diff
@@ -65,24 +65,34 @@ pub trait LinkedDataDocument {
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
 pub trait ProofSuite {
     async fn sign(
+        &self,
```

Comment:
@clehner could you please explain in the PR why we are changing from static methods to normal ones? Thanks!

Reply:
@wyc

This change is to make `ProofSuite` [object-safe](https://doc.rust-lang.org/stable/reference/items/traits.html#object-safety), i.e. so proof suites can be passed around as [trait objects](https://doc.rust-lang.org/stable/reference/types/trait-object.html).

I think this will be useful, but currently it is only used in the tests for `did-pkh`, e.g.:
https://github.com/spruceid/ssi/blob/00c7835d72b4795edf1f8eb07c69c6e2aac98b47/did-pkh/src/lib.rs#L627

The [`DIDMethod`](https://github.com/spruceid/ssi/blob/5467296/src/did.rs#L234-L241) and [`DIDResolver`](https://github.com/spruceid/ssi/blob/5467296/src/did_resolve.rs#L222-L223) traits also work this way.